cli/tests/test_repo_managed_config.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright 2025 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::common::TestEnvironment;
+
+#[test]
+fn test_repo_managed_config() {
+    let test_env = TestEnvironment::default();
+    test_env.run_jj_in(".", ["git", "init", "repo"]).success();
+    let work_dir = test_env.work_dir("repo");
+    test_env.add_config(r#"ui.pager = "user pager""#);
+
+    let output = work_dir.run_jj(["config", "get", "ui.pager"]);
+    insta::assert_snapshot!(output, @r###"
+    user pager
+    [EOF]
+    "###);
+
+    work_dir.write_file(
+        ".config/jj/config.toml",
```

Comment:
If we get close to merging this, this path should be documented in `config.md`. This probably deserves a whole section there, but it also makes sense to postpone writing it until we agree.

Reply:
Complete docs would be less critical if we disable the feature by default as I suggest in https://github.com/jj-vcs/jj/pull/6990#pullrequestreview-3035022120, though we would still want to mention *something* about this in config.md probably.

Reply:
I've now locked it behind the feature flag `repo-managed-config.enabled` and documented it in `config.md`